Automattic/sensei
# Fix patterns flicker on WPCOM

Resolves #7507
Depends on https://github.com/Automattic/wp-calypso/pull/87903

## Proposed Changes

* It updates the current logic to not display the editor welcome tour in the WPCOM editor, while displaying the Course or Lesson wizard.

### Considered alternatives:

* I tried to simulate the same behavior as the page editor on WPCOM, which also hides the welcome tour, but the actions are tied to other behaviors.
* Currently the `setShowWelcomeGuide` is trying to set the value as `undefined`, which creates a loop fetching for the state persisted in the backend.
* If we change to `false`, it works, but even with the `onlyLocal`, it's persisted in the local storage, so if we do it, it's not displayed in the other posts.
* There is a backend filter with the name `wpcom_block_editor_nux_get_status`, but it's an ajax request without context of the current page, so it's not enough for this case. (I thought some other things related to this and documented in the dependency PR, but decided on the JS hook).

## Testing Instructions


1. Checkout the dependency.
2. Build it as described in the plugin folder in the dependency repository.
3. Upload it to your atomic test site.
4. Upload the Sensei plugin with this version to your website.
5. Make sure your environment is new (you never dismissed the editor welcome tour) or force it to be displayed with `wp.data.dispatch( 'automattic/wpcom-welcome-guide' ).setShowWelcomeGuide( true )`.
6. Check that it is displayed on a normal post.
7. Check that it's not displayed in the Course and in the Lesson when the editor wizard is displayed.
8. In the wizard, navigate to the patterns step, and make sure it's not flickering anymore.
9. Check again that it continues being displayed in the normal post (it wasn't dismissed).
10. Now test the Course and Lesson wizard in a normal site (not atomic), and make sure it continues working properly.

## Deprecated Code


* `Sensei_Editor_Wizard::enqueue_admin_scripts `

## Pre-Merge Checklist
